modules/hilary/manifests/init.pp

```diff
@@ -147,7 +147,16 @@
       group   => $os_group,
       content => template('hilary/config.js.erb'),
       require => [ File[$upload_files_dir], File[$config_files_tmp_dir], File[$config_files_tmp_upload_dir] ]
-  }
+
+    # Env specific config file
+    "${app_root_dir}/${environment}.js":
+      ensure  => present,
+      mode    => "0644",
+      owner   => $os_user,
+      group   => $os_group,
+      content => template('hilary/${environment}.js'),
```

Comment:
Templates are used to process dynamic information however this uses a static template for each environment.
It would be better to either;
 Only have 1 template and use a common.json setting.
 Or have multiple files and use 'source' rather than content...
    source => "puppet:///modules/hilary/${environment}.js"
    In the puppet config those environment files would need to be in modules/hilary/files/

Reply:
Like `source => File('hilary/${environment}.js');`?

Reply:
Fixed; I also took out the extra `require`.